lib/chartlib.sh

```diff
@@ -289,11 +289,30 @@ chartlib::install_chart_with_single_config() {
             helm install "$chart_dir" --name "$release" --namespace "$namespace" --wait --timeout "$TIMEOUT"
         fi
 
+        local error=
+
         # For deployments --wait may not be sufficient because it looks at 'maxUnavailable' which is 0 by default.
         for deployment in $(kubectl get deployment --namespace "$namespace" --output jsonpath='{.items[*].metadata.name}'); do
             kubectl rollout status "deployment/$deployment" --namespace "$namespace"
+
+            # 'kubectl rollout status' does not return a non-zero exit code when rollouts fail.
+            # We, thus, need to double-check here.
+
+            local jsonpath='{.status.conditions[?(@.type=="Ready")].status}'
+
+            for pod in $(chartlib::get_pods_for_deployment "$deployment" "$namespace"); do
+                ready=$(kubectl get pod "$pod" --namespace "$namespace" --output jsonpath="$jsonpath")
+                if [[ "$ready" != "True" ]]; then
+                    chartlib::error "Pod '$pod' did not reach ready state!"
+                    error=true
```

Comment:
why not just `return 1` here, or is your idea to enumerate all the pods in all the deployments that don't go to running?

Reply:
Yes, the idea was to enumerate the pods.

Reply:
I like this much better than grepping output for `error` 👍